src/main.js

```diff
@@ -434,15 +477,20 @@ function loadKnowledgeAppContent(app, knowledgeSearchProxy, uri, contentType, ti
     context.set_timestamp(timestamp);
 
     if (!knowledgeSearchProxy) {
-        app.launch([], context);
+        log('Without a known search proxy, app ' + app.get_id() + ' cannot be launched directly');
         return;
     }
 
     knowledgeSearchProxy.LoadItemRemote(uri, '', timestamp, function(result, excp) {
         if (!excp)
             return;
         logError(excp, 'Could not load app with article ' + uri + ' fallback to just launch the app, trace');
-        app.launch([], context);
+        knowledgeSearchProxy.LaunchApplicationDirectlyRemote(timestamp, function(result, excp) {
```

Comment:
I'd rather not hide the fact that we're are launching the app indirectly, through the custom shell dbus interface.

Reply:
Fair enough. I'll update the log message.